<launch>
  <!-- start: arguments -->
  <arg name="video_path" />
  <arg name="video_title" />
  <!-- end: arguments -->

  <arg name="audio_topic_name" default="/audio" />
  <arg name="audio_channels" default="1" />
  <arg name="audio_sample_rate" default="16000" />
  <arg name="audio_format" default="wave" />
  <arg name="audio_sample_format" default="S16LE" />
  <arg name="video_topic_name" default="/head_camera_fetch_recorder/rgb/image_rect_color" />
  <arg name="video_height" default="480" />
  <arg name="video_width" default="640" />
  <arg name="video_framerate" default="30" />
  <arg name="video_encoding" default="RGB" />

  <node name="$(anon audio_video_recorder)" pkg="audio_video_recorder"
        type="audio_video_recorder" output="screen">
    <remap from="~input/audio" to="$(arg audio_topic_name)" />
    <remap from="~input/image" to="$(arg video_topic_name)" />
    <rosparam subst_value="true">
      queue_size: 100
      file_name: $(arg video_path)/$(arg video_title)
      file_format: avi
      audio_channels: $(arg audio_channels)
      audio_sample_rate: $(arg audio_sample_rate)
      audio_format: $(arg audio_format)
      audio_sample_format: $(arg audio_sample_format)
      video_height: $(arg video_height)
      video_width: $(arg video_width)
      video_framerate: $(arg video_framerate)
      video_encoding: $(arg video_encoding)
    </rosparam>
  </node>
</launch>
